stdlib/os/path.pyi

```diff
@@ -77,21 +78,30 @@ else:
         @overload
         def realpath(filename: AnyStr) -> AnyStr: ...
 
-# In reality it returns str for sequences of StrPath and bytes for sequences
-# of BytesPath, but mypy does not accept such a signature.
-def commonpath(paths: Sequence[AnyPath]) -> Any: ...
+@overload
+def commonpath(paths: Sequence[StrPath]) -> str: ...
+@overload
+def commonpath(paths: Sequence[BytesPath]) -> bytes: ...
 
-# NOTE: Empty lists results in '' (str) regardless of contained type.
-# So, fall back to Any
-def commonprefix(m: Sequence[AnyPath]) -> Any: ...
-def lexists(path: AnyPath) -> bool: ...
+# All overloads can return empty string. Ideally, Literal[""] would be a valid
+# Iterable[_T], so that Union[List[_T], Literal[""]] could be used as a return
+# type. But because _T is not necessarily str, we need Sequence[_T] instead.
+@overload
+def commonprefix(m: Sequence[StrPath]) -> str: ...
+@overload
+def commonprefix(m: Sequence[BytesPath]) -> Union[bytes, Literal[""]]: ...
+@overload
+def commonprefix(m: Sequence[List[_T]]) -> Sequence[_T]: ...
```

Comment:
Shouldn't there be a bound on `_T`?

Reply:
`commonprefix([[1, 2], [1, 3]]) == [1]`. It is weird, but it works. Should we still put a bound?

Reply:
Oh wow, you're right. I think there should be a bound of `SupportsLessThan` because it uses `min` and `max` internally. Also, the tuple overload below needs `, ...`.